tailfs/tailfsimpl/dirfs/mkdir.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package dirfs
+
+import (
+	"context"
+	"os"
+
+	"tailscale.com/tailfs/tailfsimpl/shared"
+)
+
+// Mkdir implements webdav.FileSystem. All attempts to Mkdir a directory that
+// already exists will succeed. All other attempts will fail with
+// os.ErrPermission.
+func (dfs *FS) Mkdir(ctx context.Context, name string, perm os.FileMode) error {
+	nameWithoutStaticRoot, isStaticRoot := dfs.trimStaticRoot(name)
+	if isStaticRoot || shared.IsRoot(name) {
+		// root directory already exists, consider this okay
+		return nil
+	}
+
+	child := dfs.childFor(nameWithoutStaticRoot)
+	if child != nil {
+		// child already exists, consider this okay
+		return nil
+	}
+
+	return os.ErrPermission
```

Comment:
I guess if we want these to look like os package errors we could also:

```go
return &os.PathError{Op: "mkdir", Path: name, Err: os.ErrPermission}
```

if that matters. (It's nice when logging, but maybe not otherwise that critical)